### PR TITLE
Fix no internal stats regression

### LINKED ISSUE #316: #290 introduces flusher bug that causes internal metrics to not be sent
CC @ash2k 

We just upgraded our production release from 20.3.1 to 22.0.0 and noticed that we stopped sending internal metrics about backend events, specifically: https://github.com/atlassian/gostatsd/blob/979ab8493471cbea7625d07fcf48600209bf45a3/pkg/backends/datadog/datadog.go#L146-L151

A debug session shows that the `flushed` channel is never called on a flush.

I did a git bisect and found the offending commit is from https://github.com/atlassian/gostatsd/pull/290

### PR DESCRIPTION
**Pull request by ash2k**
Fixes #316.

### CODE REVIEW

File: cmd/gostatsd/main.go [hunk 1]
```diff
@@ -59,23 +58,17 @@ func main() {
...
+	s, err := constructServer(v)
 	if err != nil {
 		return err
 	}
 
-	stgr := stager.New()
-	defer stgr.Shutdown()
-	for _, runnable := range runnables {
-		stgr.NextStage().StartWithContext(runnable)
+	profileAddr := v.GetString(ParamProfile)
```
Comment (ash2k) [inline cmd/gostatsd/main.go:67]: Moved it here - first try to construct the server and only then start profiler.

**Comment by tiedotguy** [comment]
Pushed a changelog, looks good to me.  Hopefully @hligit's change works, because it's going out :)

**Comment by tiedotguy** [comment]
Sigh. I guess coveralls may need to go.

**Comment by hligit** [comment]
> Pushed a changelog, looks good to me. Hopefully @hligit's change works, because it's going out :)

Confirm the change of #310 works in my lab environment.

**Pull request merged**